### PR TITLE
Ajio Assignment

### PR DESCRIPTION
**Pull request by MedhaGupta-Ril**

### CODE REVIEW

File: src/test/java/com/znsio/sample/e2e/screen/ajio/SearchResultsScreen.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public static SearchResultsScreen get() {
+        Driver driver = fetchDriver(Thread.currentThread()
+                                          .getId());
+        Platform platform = Runner.fetchPlatform(Thread.currentThread()
+                                                       .getId());
+        LOGGER.info(SCREEN_NAME + ": Driver type: " + driver.getType() + ": Platform: " + platform);
+        Visual visually = fetchEyes(Thread.currentThread()
+                                          .getId());
+
+        switch(platform) {
```
Comment (Mohammed9901) [inline src/test/java/com/znsio/sample/e2e/screen/ajio/SearchResultsScreen.java:30]: Remove instance for android as you have not used in scenario.

File: src/test/java/com/znsio/sample/e2e/screen/web/ajio/ProductDetailScreenWeb.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        this.visually = visually;
+    }
+
+    @Override
+    public ProductDetailScreen selectProductSize(){
+        driver.switchToNextTab();
+        visually.checkWindow(SCREEN_NAME, "Product detail screen");
+        WebElement webElement = driver.waitTillElementIsPresent(byProductSizeXpath);
+        LOGGER.info(String.format("Product size selected '%s'", webElement.getText()));
+        webElement.click();
```
Comment (Mohammed9901) [inline src/test/java/com/znsio/sample/e2e/screen/web/ajio/ProductDetailScreenWeb.java:32]: Directly perform the action without creating an instance.

File: src/test/java/com/znsio/sample/e2e/screen/web/ajio/ProductDetailScreenWeb.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        this.driver = driver;
+        this.visually = visually;
+    }
+
+    @Override
+    public ProductDetailScreen selectProductSize(){
+        driver.switchToNextTab();
+        visually.checkWindow(SCREEN_NAME, "Product detail screen");
+        WebElement webElement = driver.waitTillElementIsPresent(byProductSizeXpath);
+        LOGGER.info(String.format("Product size selected '%s'", webElement.getText()));
```
Comment (Mohammed9901) [inline src/test/java/com/znsio/sample/e2e/screen/web/ajio/ProductDetailScreenWeb.java:31]: Avoid using unnecessary loggers, if you have provided assertions in BL.

File: src/test/java/com/znsio/sample/e2e/screen/web/ajio/ProductDetailScreenWeb.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class ProductDetailScreenWeb extends ProductDetailScreen {
+    private final Driver driver;
+    private final Visual visually;
+    private static final String SCREEN_NAME = ProductDetailScreenWeb.class.getSimpleName();
+    private static final Logger LOGGER = Logger.getLogger(SCREEN_NAME);
+
+    //improve xpath
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/screen/web/ajio/ProductDetailScreenWeb.java:16]: remove unwanted comments

File: src/test/java/com/znsio/sample/e2e/screen/android/ajio/ShoppingCartScreenAndroid.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import org.apache.log4j.Logger;
+import org.openqa.selenium.By;
+
+public class ShoppingCartScreenAndroid extends ShoppingCartScreen {
+
+    private final Driver driver;
+    private final Visual visually;
+    private static final String SCREEN_NAME = ShoppingCartScreenAndroid.class.getSimpleName();
+    private static final Logger LOGGER = Logger.getLogger(SCREEN_NAME);
+    private static final String NOT_YET_IMPLEMENTED = " not yet implemented";
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/screen/android/ajio/ShoppingCartScreenAndroid.java:15]: remove unused variables

File: src/test/java/com/znsio/sample/e2e/screen/android/ajio/ShoppingCartScreenAndroid.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    private static final By byProductBrandId = By.id("com.ril.ajio:id/brandInfo");
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/screen/android/ajio/ShoppingCartScreenAndroid.java:16]: package name can be removed

File: src/test/java/com/znsio/sample/e2e/screen/android/ajio/ProductDetailScreenAndroid.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+public class ProductDetailScreenAndroid extends ProductDetailScreen {
+
+    private final Driver driver;
+    private final Visual visually;
+    private static final String SCREEN_NAME = ProductDetailScreenAndroid.class.getSimpleName();
+    private static final Logger LOGGER = Logger.getLogger(SCREEN_NAME);
+    private static final String NOT_YET_IMPLEMENTED = " not yet implemented";
+    private static final By byProductBrandId = By.id("com.ril.ajio:id/brand_name");
+    private static final By byAddToCartId = By.id("com.ril.ajio:id/add_to_cart_tv");
+    private static final By byProductSizeXpath = By.xpath("//android.widget.TextView[@resource-id = 'com.ril.ajio:id/row_pdp_fixed_size_tv' and @text = '8']");
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/screen/android/ajio/ProductDetailScreenAndroid.java:18]: try to use better locators

File: src/test/java/com/znsio/sample/e2e/screen/web/ajio/AjioHomeScreenWeb.java [hunk 8]
```diff
@@ -25,14 +25,14 @@ public AjioHomeScreenWeb(Driver driver, Visual visually) {
     }
 
     @Override
-    public AjioSearchResultsScreen searchFor(String product) {
+    public SearchResultsScreen searchFor(String product) {
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/screen/web/ajio/AjioHomeScreenWeb.java:28]: Class Name Can be HomeScreenWeb as package name already contains ajio

**Review by vishal07111998 — CHANGES_REQUESTED** [review]

**Comment by vishal07111998** [comment]
Remove unwanted data, files or directories that you are not using

**Pull request closed without merging**